modules/openid_auth/app/controllers/openid_auth/v0/mvi_users_controller.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+require_dependency 'openid_auth/application_controller'
+
+module OpenidAuth
+  module V0
+    class MviUsersController < ApplicationController
+      skip_before_action :authenticate
+      before_action :check_required_headers
+      before_action :build_identity
+
+      def show
+        if request.headers['x-va-ssn'].present?
+          build_ssn_user
+        else
+          build_edipi_user
+        end
+        service = MVI::Service.new
+        @mvi_response = service.find_profile(@user)
+        if @mvi_response.profile.try(:icn).present?
+          icn_found
+        else
+          raise Common::Exceptions::RecordNotFound, 'ICN Not Found'
+        end
+      end
+
+      private
+
+      def build_edipi_user
+        @user = User.create(uuid: request.headers['x-va-edipi'],
+                            email: request.headers['x-va-user-email'],
+                            ssn: request.headers['x-va-edipi'],
+                            loa:
+                            {
+                              current: request.headers['x-va-level-of-assurance'].to_i,
+                              highest: highest_loa
+                            })
+      end
+
+      def build_ssn_user
+        @user = User.create(uuid: request.headers['x-va-ssn'],
+                            email: request.headers['x-va-user-email'],
+                            first_name: request.headers['x-va-first-name'],
+                            last_name: request.headers['x-va-last-name'],
+                            birth_date: request.headers['x-va-dob'],
+                            ssn: request.headers['x-va-ssn'],
+                            loa:
+                            {
+                              current: request.headers['x-va-level-of-assurance'].to_i,
+                              highest: highest_loa
+                            })
+      end
+
+      def icn_found
+        render json:
+          {
+            "id": @mvi_response.profile.icn,
+            "type": 'user-mvi-icn',
+            "data": {
+              "attributes": {
+                "icn": @mvi_response.profile.icn
+              }
+            }
+          }
+      end
+
+      def highest_loa
+        request.headers['x-va-level-of-assurance'].to_i == 3 ? 3 : 1
+      end
+
+      def check_required_headers
+        raise Common::Exceptions::ParameterMissing, 'X-VA-SSN or X-VA-EDIPI' if missing_ssn_or_edipi
+        raise Common::Exceptions::ParameterMissing, 'x-va-level-of-assurance' if missing_loa
+        raise Common::Exceptions::ParameterMissing, 'x-va-user-email' if missing_email
+      end
```

Comment:
Should this controller be emitting a `Vary` header to ensure proper caching? I don't know of anything between saml-proxy and vets-api but I'd hate to put some sort of caching proxy between them in the future and introduce a bug.